src/m_menu.c

```diff
@@ -5334,7 +5334,7 @@ boolean M_Responder (event_t* ev)
 	{
 	  if (saveCharIndex > 0)
 	    {
-	      if (StartsWithMapIdentifier(savegamestrings[saveSlot]))
+	      if (StartsWithMapIdentifier(savegamestrings[saveSlot]) && saveStringEnter == 1)
 	        saveCharIndex = 0;
```

Comment:
I suggest to remove this `if ...` code block and leave only `saveCharIndex--`, I think it's more consistent and convenient.

Reply:
> I suggest to remove this if ... code block and leave only saveCharIndex--, I think it's more consistent and convenient.

So we always remove only the last char?

Reply:
> So we always remove only the last char?

Yes, this is what I expected, I like the consistency in the UI. To be honest, I've never edited a save name before 😄

Reply:
> To be honest, I've never edited a save name before 😄

Me neither. 😁 But I, too, would be shocked to find my entire save string removed when I press backspace while editing it, just because it *starts* with a map name.